pkg/tsdb/tempo/grpc.go

```diff
@@ -74,6 +76,39 @@ func getDialOpts(settings backend.DataSourceInstanceSettings, opts httpclient.Op
 		dialOps = append(dialOps, grpc.WithTransportCredentials(insecure.NewCredentials()))
 	}
 
+	// The following code is required to make gRPC work with Grafana Cloud PDC
+	// (https://grafana.com/docs/grafana-cloud/connect-externally-hosted/private-data-source-connect/)
+	proxyClient, err := settings.ProxyClient(ctx)
+	if err != nil {
+		logger.Error("Error getting proxy client. The Tempo plugin will not use the secure socks proxy", "error", err)
+		return dialOps
+	}
+	if proxyClient.SecureSocksProxyEnabled() { // secure socks proxy is behind a feature flag
+		dialer, err := proxyClient.NewSecureSocksProxyContextDialer()
+		if err != nil {
+			logger.Error("Error dialing secure socks proxy. The Tempo plugin will not use the secure socks proxy", "error", err)
+			return dialOps
+		}
+		logger.Debug("gRPC dialer instantiated. Appending gRPC dialer to dial options")
+		dialOps = append(dialOps, grpc.WithContextDialer(func(ctx context.Context, host string) (net.Conn, error) {
+			select {
+			case <-ctx.Done():
+				logger.Warn("Context has been canceled, this should not happen. Proceeding with dialing anyway")
```

Comment:
This does not seem right and would expect some comment here explaining why we think it's ok to proceed dialing if the context is canceled and why it should not happen. I assume cancelation can happen for any reason that is outside of the competency of this code that is why contexts are propagated.

Reply:
Yeah I did this but to be honest I'm not sure what and how exactly we should here, so I didn't want to block the flow on a potentially wrong check but just have some logs for future troubleshooting. 
I can simply remove this for now and improve it if we have issues.

Reply:
I don't understand why you decided to remove the code. I mean don't know the code that well but I am not saying we should not consider context cancelation. If there is something unclear it makes sense to find out how things work instead just adding or removing code blindly. In this case maybe it make sense to check for cancelation and return some sort of cancelation error if context is already canceled. It definitely does not seem to make sense to dial connection if its not actually going to be used.

Reply:
I'm still off, so answering quickly: I improved the check on the context and I plan to monitor this when we release this incrementally to customers. 
I don't know if there is a way to trigger context cancellation in cloud to validate the error management, I'd be happy to know about this, because I'm not sure on how to test this, which is probably was this caused confusion and you thought I was blindly adding code.

Reply:
Discussed offline.